### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/exiv2/commit/70a7c49f45456c86e511644b3108242566a598a2#commitcomment-65911822

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/exiv2/commit/37f930d9eb667c61a98c4307011a509c329cb6bb

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I intend to annotate this PR shortly. I'm happy to drop changes to files, drop specific terms, or split things as requested, although this may take some time.

### CODE REVIEW

**Review by sridharb1 — DISMISSED** [review]
Thank you, looks good

File: exiv2.md [hunk 1]
```diff
@@ -1008,8 +1008,8 @@ HH:MM:SScHH:MM
 ```
 where HH:MM:SS refers to local time's hour, minutes and seconds and, HH:MM 
 refers to hours and minutes with c being ahead (+) or behind (-) 
-Universal Coordinated Time (UCT) as described in ISO 8601 (e.g., 
-08:45:15+01:00 is 08:45:15, which is +1 hour different from UCT).
+Coordinated Universal Time (UTC) as described in ISO 8601 (e.g., 
+08:45:15+01:00 is 08:45:15, which is +1 hour different from UTC).
```
Comment (jsoref) [inline exiv2.md:1012]: This is probably the biggest stretch as a spelling fix. Specifically, `UCT` isn't a thing.

There are 18x as many hits for `"Coordinated Universal Time"` than `"Universal Coordinated Time"`, and the top hits for the latter show the former.

File: include/exiv2/types.hpp [hunk 2]
```diff
@@ -102,12 +102,12 @@ namespace Exiv2 {
         asciiString        = 2, //!< Exif ASCII type, 8-bit byte.
         unsignedShort      = 3, //!< Exif SHORT type, 16-bit (2-byte) unsigned integer.
         unsignedLong       = 4, //!< Exif LONG type, 32-bit (4-byte) unsigned integer.
-        unsignedRational   = 5, //!< Exif RATIONAL type, two LONGs: numerator and denumerator of a fraction.
+        unsignedRational   = 5, //!< Exif RATIONAL type, two LONGs: numerator and denominator of a fraction.
```
Comment (jsoref) [inline include/exiv2/types.hpp:105]: Sometimes I run into projects that are trying to do something special here. But I think this is right.

File: po/exiv2.pot [hunk 3]
```diff
@@ -13401,7 +13401,7 @@ msgstr ""
 
 #: src/properties.cpp:277
 msgid ""
-"References to resources that were incorporated, byinclusion or reference, "
+"References to resources that were incorporated, by inclusion or reference, "
```
Comment (jsoref) [inline po/exiv2.pot:13404]: I've mostly ignored the `.pot` file (and probably should have ignored it entirely).

File: releasenotes/releasenotes.txt [hunk 4]
```diff
@@ -25,10 +25,10 @@ Thank You to the following who have contributed to this release.
 1) Bug and security fixes:
    - bmff read-only support (.CR3, .HEIC, .HEIF, .AVIF and JXL/bmff).
    - Security process is defined.  https://github.com/Exiv2/exiv2/security/policy
-   - 3 benign ASAN issues.  First CVE in 18 months.  Patched in 2hrs, Fixed ixed in 24hrs.
+   - 3 benign ASAN issues.  First CVE in 18 months.  Patched in 2hrs, Fixed in 24hrs.
```
Comment (jsoref) [inline releasenotes/releasenotes.txt:28]: Interesting stray near duplicate corruption.

File: releasenotes/releasenotes.txt [hunk 5]
```diff
@@ -25,10 +25,10 @@ Thank You to the following who have contributed to this release.
...
    - DNG 1.6 and Exif 2.32 support.
    - More easyaccess APIs and documentation.
-   - Localation support on Crowdin.
+   - Localization support on Crowdin.
    - Updated Catalan localisation.
```
Comment (jsoref) [inline releasenotes/releasenotes.txt:32]: Perhaps en-GB spelling would be better...

File: samples/conntest.cpp [hunk 6]
```diff
@@ -73,7 +73,7 @@ void curlcon(const std::string& url, bool useHttp1_0 = false) {
 
     /* Perform the request, res will get the return code */
     CURLcode res = curl_easy_perform(curl);
-    if(res != CURLE_OK) { // error happends
+    if(res != CURLE_OK) { // error happened
```
Comment (jsoref) [inline samples/conntest.cpp:76]: or `happens`?

File: src/canonmn_int.cpp [hunk 7]
```diff
@@ -740,8 +740,8 @@ namespace Exiv2 {
          {0x000a, "WB_GRGBLevelsDaylight", N_("WB_G RGB Levels Daylight"), N_("WB_G RGB Levels Daylight"), canonWbId, makerTags, unsignedLong, -1, printValue},
          {0x0012, "WB_GRGBLevelsCloudy", N_("WB_G RGB Levels Cloudy"), N_("WB_G RGB Levels Cloudy"), canonWbId, makerTags, unsignedLong, -1, printValue},
          {0x001a, "WB_GRGBLevelsTungsten", N_("WB_G RGB Levels Tungsten"), N_("WB_G RGB Levels Tungsten"), canonWbId, makerTags, unsignedLong, -1, printValue},
-         {0x0022, "WB_GRGBLevelsFluorescent", N_("WB_G RGB Levels Flourescent"), N_("WB_G RGB Levels Flourescent"), canonWbId, makerTags, unsignedLong, -1, printValue},
-         {0x002a, "WB_GRGBLevelsFluorHigh", N_("WB_G RGB Levels Flourecent High"), N_("WB_G RGB Levels Flourecent High"), canonWbId, makerTags, unsignedLong, -1, printValue},
+         {0x0022, "WB_GRGBLevelsFluorescent", N_("WB_G RGB Levels Fluorescent"), N_("WB_G RGB Levels Fluorescent"), canonWbId, makerTags, unsignedLong, -1, printValue},
+         {0x002a, "WB_GRGBLevelsFluorHigh", N_("WB_G RGB Levels Fluorescent High"), N_("WB_G RGB Levels Fluorescent High"), canonWbId, makerTags, unsignedLong, -1, printValue},
```
Comment (jsoref) [inline src/canonmn_int.cpp:744]: I'm fairly certain these are user facing localization elements...

File: src/casiomn_int.cpp [hunk 8]
```diff
@@ -369,7 +369,7 @@ namespace Exiv2 {
     constexpr TagDetails casio2AutoISO[] = {
         {  1, N_("On")                   },
         {  2, N_("Off")                  },
-        {  7, N_("On (high sensitiviy)") },
+        {  7, N_("On (high sensitivity)") },
         {  8, N_("On (anti-shake)")      },
         { 10, N_("High Speed")           }
```
Comment (jsoref) [inline src/casiomn_int.cpp:374]: I suppose I could add a space for each of the other rows

File: src/crwimage_int.cpp [hunk 9]
```diff
@@ -1103,7 +1103,7 @@ namespace Exiv2 {
         assert(ifdId != ifdIdNotSet);
         DataBuf buf = packIfdId(image.exifData(), ifdId, pHead->byteOrder());
         if (buf.size() == 0) {
-            // Try the undecoded tag
+            // Try the raw tag
```
Comment (jsoref) [inline src/crwimage_int.cpp:1106]: This is probably the biggest change in the PR. I'm not sure about it.

File: src/nikonmn_int.cpp [hunk 10]
```diff
@@ -61,7 +61,7 @@ namespace Exiv2 {
     };
 
     //! Off, Low, Normal, High, multiple tags
-    constexpr TagDetails nikonActiveDLighning[] = {
+    constexpr TagDetails nikonActiveDLighting[] = {
```
Comment (jsoref) [inline src/nikonmn_int.cpp:64]: moderately confident this is right

File: src/nikonmn_int.cpp [hunk 11]
```diff
@@ -1678,12 +1678,12 @@ namespace Exiv2 {
             // Mapping by Roger Larsson
             unsigned focusmetering = value.toLong(0);
             unsigned focuspoint = value.toLong(1);
-            unsigned focusused = (value.toLong(2) << 8) + value.toLong(3);
+            unsigned focussed = (value.toLong(2) << 8) + value.toLong(3);
```
Comment (jsoref) [inline src/nikonmn_int.cpp:1681]: This could be wrong if it's `focus` + `used` instead of `focussed`.

File: src/panasonicmn_int.cpp [hunk 12]
```diff
@@ -601,7 +601,7 @@ namespace Exiv2 {
 
     } // PanasonicMakerNote::print0x0029
 
-    // bage age
+    // baby age
```
Comment (jsoref) [inline src/panasonicmn_int.cpp:604]: I had to look this one up...

File: xmpsdk/include/TXMPUtils.hpp [hunk 13]
```diff
@@ -769,7 +769,7 @@ template <class tStringObj> class TXMPUtils {
 									 XMP_StringPtr                separator,
 									 XMP_StringPtr                quotes,
 									 XMP_OptionBits               options,
-									 tStringObj *                 catedStr );
+									 tStringObj *                 concatenatedStr );
```
Comment (jsoref) [inline xmpsdk/include/TXMPUtils.hpp:772]: I don't see any particular reason to shorten this token, but I'm happy to drop it (or any other change(s)) if requested.

File: xmpsdk/include/client-glue/TXMPMeta.incl_cpp [hunk 14]
```diff
@@ -90,7 +90,7 @@ Terminate()
 }
 
 // =================================================================================================
-// Constuctors, destructor, operators
+// Constructors, destructor, operators
 // ==================================
```
Comment (jsoref) [inline xmpsdk/include/client-glue/TXMPMeta.incl_cpp:94]: Here and below are places where apparently the bottom `=`/`-` should be as long as the text...

File: xmpsdk/src/XMPUtils-FileInfo.cpp [hunk 15]
```diff
@@ -210,11 +210,11 @@ ClassifyCharacter ( XMP_StringPtr fullString, size_t offset,
 
 
 // -------------------------------------------------------------------------------------------------
-// IsClosingingQuote
+// IsClosingQuote
 // -----------------
```
Comment (jsoref) [inline xmpsdk/src/XMPUtils-FileInfo.cpp:214]: `-`....

**Review by jsoref — COMMENTED** [review]
The vast majority of the corrections were automatically suggested by Google Sheets (especially any preference for en-US over en-GB).

But all fault is mine.

File: doc/ChangeLog [hunk 16]
```diff
@@ -323,7 +323,7 @@ Build: (11)
 	0001329	Error during cmake with gcc version parsing
 	0001310	Provide support for msys/2.0
 	0001299	exiv2-0.26-trunk.tar.gz changed on download server
-	0001270	Using libexiv2.a/.lib  in multhreaded app segfaults.
+	0001270	Using libexiv2.a/.lib  in multithreaded app segfaults.
```
[the remark this answers is not in the document]
Comment (piponazo) [reply doc/ChangeLog:326]: It is fine for me to fix typos in the changlog too 👍

File: doc/ChangeLog [hunk 17]
```diff
@@ -643,9 +643,9 @@ Changes from version 0.25 to 0.26
 	- 0001145:	Respect Sony/Minolta lenses with shared LensID such as Tamron SP AF 17-50mm F2.8 XR Di II LD
 	- 0001144:	Sigma 10-20mm f/4-5.6 EX DC is detected as Tamaron
 			(Simon Harhues)
-	- 0001142:	Manual lens does not get recogniced: Beroflex zoom 500mm
+	- 0001142:	Manual lens does not get recognized: Beroflex zoom 500mm
```
[the remark this answers is not in the document]
Comment (piponazo) [reply doc/ChangeLog:646]: I would wait for @clanmills feedback on this matter. He has been the main maintainer of the project for many years, and I am sure he will have something to say about this 😸.

I do not have a strong preference for one or another.

File: xmpsdk/include/TXMPUtils.hpp (hunk 13 above)
Comment (piponazo) [reply xmpsdk/include/TXMPUtils.hpp:772]: I would vote to drop all the changes done inside the **xmpsdk** folder. This is a 3rd party dependency in the project that was included in our codebase a long time ago, but we try to not touch these files too much.

**Review by piponazo — CHANGES_REQUESTED** [review]
Thanks a lot for the contribution! It is really appreciated.

I would also love to see (probably in a separated PR) the inclusion on the action in the project. Most of the collaborators are non native English-speakers, and it will probably find us to help typos eventually.

I made few comments here and there about things we could drop.

**Comment by sridharb1** [comment]
Yes, Luis, I also thought about the focus used making sense, but I also thought that focused also makes sense, as in, if it is focused, then focus is used.




On Thursday, February 3, 2022, 13:35, Luis Díaz Más ***@***.***> wrote:




@piponazo requested changes on this pull request.

Thanks a lot for the contribution! It is really appreciated.

I would also love to see (probably in a separated PR) the inclusion on the action in the project. Most of the collaborators are non native English-speakers, and it will probably find us to help typos eventually.

I made few comments here and there about things we could drop.

In po/exiv2.pot:
> -"References to resources that were incorporated, byinclusion or reference, "
+"References to resources that were incorporated, by inclusion or reference, "

I think it is fine, since you also fixed the typo in the properties.cpp file.

In src/crwimage_int.cpp:
> -            // Try the undecoded tag
+            // Try the raw tag

Right ... I am not very sure about what would be the right word to use there. I would vote to drop this change.

In src/nikonmn_int.cpp:
> -            unsigned focusused = (value.toLong(2) << 8) + value.toLong(3);
+            unsigned focussed = (value.toLong(2) << 8) + value.toLong(3);

Hehe interesting ... I think this should be dropped because it problably means "FocusUsed". As you can see between lines 1721-1733, we are trying to determine if a focus point was used (and which one) or not. This correspond to AFPointsInFocus:
https://www.exiftool.org/TagNames/Nikon.html#AFInfo

In xmpsdk/include/TXMPUtils.hpp:
> -									 tStringObj *                 catedStr );
+									 tStringObj *                 concatenatedStr );

I would vote to drop all the changes done inside the xmpsdk folder. This is a 3rd party dependency in the project that was included in our codebase a long time ago, but we try to not touch these files too much.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by clanmills** [comment]
@sridharb1 Thank You for saying _"I would vote to drop all the changes done inside the xmpsdk folder. "_.  I agree and you've explained the reason.

@jsoref This is a really good contribution.  I'm a native english speaker (Scottish) and I'm dyslexic (also Scottish).  So all the typos in README.md and elsewhere are mine.  Thank You for working on this.  I prefer en-US because it's more conventional for Open-Source. _(I'm also an US Citizen)_.

When you've finished on this, I'd appreciate similar treatment being given to my book:  https://github.com/Exiv2/team.exiv2.org

File: README.md [hunk 18]
```diff
@@ -6,7 +6,7 @@
...
+| [![Basic CI for all platforms on push](https://github.com/Exiv2/exiv2/actions/workflows/on_push_BasicWinLinMac.yml/badge.svg?branch=main)](https://github.com/Exiv2/exiv2/actions/workflows/on_push_BasicWinLinMac.yml) |  [![CI for different Linux distributions](https://github.com/Exiv2/exiv2/actions/workflows/nightly_Linux_distributions.yml/badge.svg?branch=main)](https://github.com/Exiv2/exiv2/actions/workflows/nightly_Linux_distributions.yml) |  [![Linux Special Builds on PRs](https://github.com/Exiv2/exiv2/actions/workflows/on_PR_linux_special_builds.yml/badge.svg)](https://github.com/Exiv2/exiv2/actions/workflows/on_PR_linux_special_builds.yml) |
```
[the remark this answers is not in the document]
Comment (kmilos) [reply README.md:9]: This is fine, but note this will lead to a completely new workflow created [here](https://github.com/Exiv2/exiv2/actions) (with the same name though), and [the old runs will have to be deleted via the API](https://github.com/Exiv2/exiv2/pull/2049#issuecomment-1006710079).

File: include/exiv2/datasets.hpp [hunk 19]
```diff
@@ -187,7 +187,7 @@ namespace Exiv2 {
          */
         static const char* dataSetDesc(uint16_t number, uint16_t recordId);
         /*!
-          @brief Return the photohsop name of a given dataset.
+          @brief Return the photoshop name of a given dataset.
```
Comment (kmilos) [inline include/exiv2/datasets.hpp:190]: Photoshop should also be capitalized (as a brand name)? See also line 193 below (and look for other instances)?

**Review by piponazo — APPROVED** [review]
